src/Linters/Linter.hack

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+/**
+ * The process to find problems against a single source file.
+ *
+ * Problems found by a Linter could associated with different LintRules,
+ * especially when the Linter is a proxy calling other linting tools.
+ */
+<<
+  __Sealed(
+    SingleRuleLinter::class,
+    // HHClientLinter::class
+  ),
+  __ConsistentConstruct,
+>>
+interface Linter {
+  abstract const type TConfig;
+
+  public function getProblemsAsync(): Awaitable<vec<LintError>>;
+
+  public static function shouldLintFile(File $_): bool;
+
+  public function __construct(File $file, ?this::TConfig $config);
+
+  public function isLinterSuppressedForFile(): bool;
+
+  public function getFile(): File;
+
+  static public function typeAssertConfig(mixed $config): this::TConfig;
```

Comment:
Can you experiment (in a different PR) with removing this?

Right now this uses the TypeAssert library, but that's because HHAST is older than the `is` and `as` operators, which should work for reasonable types of `this::TConfig`; we shouldn't have a method if `$value as this::TConfig` is usable.

search on github is also not showing me any callers to this?

Reply:
Will do!

Reply:
#386